wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/remote/LsConfigurationExtractor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.languageserver.remote;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.unmodifiableList;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.eclipse.che.api.languageserver.registry.DocumentFilter;
+import org.eclipse.che.api.languageserver.registry.LanguageServerDescription;
+
+/**
+ * This class is responsible for language server description extraction out of server attributes
+ * map. It is expected that there will be specific attribute named <code>config</code> that will
+ * contain serialized json data that represents all language server configuration data. Structure of
+ * json corresponds to {@link LanguageServerDescription} class with all aggregated classes
+ */
+@Singleton
+class LsConfigurationExtractor {
```

Comment:
This looks much more complex than parsing to an object. Have you considered creating a model object?

Reply:
Yes, I had some consideration for creating a model object.